TicketsDemo.Domain/DefaultImplementations/WeekendsAndHolidaysPriceCalculationStrategy.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+using TicketsDemo.Domain.Interfaces;
+using TicketsDemo.XML;
+
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    public class WeekendsAndHolidaysPriceCalculationStrategy : IPriceCalculationStrategy
+    {
+        private IHolidayRepository _holidayRepository;
+        
+        public WeekendsAndHolidaysPriceCalculationStrategy(IHolidayRepository holidayRepository)
+        {
+            _holidayRepository = holidayRepository;
+        }
+
+        public List<PriceComponent> CalculatePrice(PlaceInRun placeInRun)
+        {
+            var components = new List<PriceComponent>();
+            var runDate = placeInRun.Run.Date;
+            foreach (Holiday holiday in _holidayRepository.GetHolidaysList())
+            {
+                if (runDate.Day == holiday.Date.Day && runDate.Month == holiday.Date.Month && runDate.Year == holiday.Date.Year)
+                {
+                    var HolidayComponent = new PriceComponent()
+                    {
+                        Name = $"Holiday service tax for {holiday.Name}",
+                        Value = 10m
```

Comment:
holiday and weekend price components are static, which is wrong in terms of real prices of railway tickets (which can be 10hrn to 2000hrn based on the train/carriage type).

This strategy shall receive another IPriceCalculationStrategy instance using DI and use retreived price components as base number. Holiday/weekend overhead is then to be added as % from this base number